File: typescript/test/utils.test.ts
import { describe, expect, test } from 'vitest';
import { isSquarePositiveInteger } from '../src/utils.js';

describe('Utils', () => {
  describe('isSquarePositiveInteger', () => {
    test('Returns false for negative number', () => {
      expect(isSquarePositiveInteger(-4)).toBe(false);
    });

    test('Returns false for non-integer number', () => {
      expect(isSquarePositiveInteger(2.5)).toBe(false);
    });

    test('Returns false for non-square integer', () => {
      expect(isSquarePositiveInteger(3)).toBe(false);
    });

    test('Returns true for square postive integer', () => {
      expect(isSquarePositiveInteger(9)).toBe(true);
    });

    test('Returns true for very large squares', () => {
      expect(isSquarePositiveInteger(2 ** (2 ** (2 ** 2)))).toBe(true);
    });
  });
});

File: kotlin/src/test/kotlin/UtilsTest.kt
import dev.patientallison.sudoku.isPositive
import dev.patientallison.sudoku.isPositiveSquare
import org.junit.jupiter.api.Nested
import kotlin.test.Test
import kotlin.test.assertFalse
import kotlin.test.assertTrue

class UtilsTest {
    @Nested
    inner class IsPositive {
        @Test
        fun `returns true for positive integer`() {
            assertTrue(isPositive(5))
        }

        @Test
        fun `returns false for zero`() {
            assertFalse(isPositive(0))
        }

        @Test
        fun `returns false for negative`() {
            assertFalse(isPositive(-1))
        }
    }

    @Nested
    inner class IsPositiveSquare {
        @Test
        fun `returns true for positive square`() {
            assertTrue(isPositiveSquare(9))
        }

        @Test
        fun `returns false for 0`() {
            assertFalse(isPositiveSquare(0))
        }

        @Test
        fun `returns false for negative square`() {
            assertFalse(isPositiveSquare(-9))
        }

        @Test
        fun `returns false for non-square`() {
            assertFalse(isPositiveSquare(3))
        }
    }
}